packages/react-devtools-shared/src/devtools/views/Components/EditableValue.js

```diff
@@ -27,20 +28,38 @@ export default function EditableValue({
   path,
   value,
 }: EditableValueProps) {
+  const [isValid, setIsValid] = useState(true);
   const [hasPendingChanges, setHasPendingChanges] = useState(false);
-  const [editableValue, setEditableValue] = useState(value);
+  const [stringifiedValue, setStringifiedValue] = useState(
+    JSON.stringify(value),
+  );
+  const [editableValue, setEditableValue] = useState(stringifiedValue);
   const inputRef = useRef<HTMLInputElement | null>(null);
 
-  if (hasPendingChanges && editableValue === value) {
+  useEffect(
```

Comment:
If the value changes we need to update the stringified version of it.

Reply:
Looks like this?

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-unconditionally-copying-props-to-state

Reply:
Rather, that looks like it should be a `useMemo`, or even not a hook at all. You call `JSON.stringify` for the (unused) initializer value on every render pass anyway.

Reply:
I completely removed this bit. Thanks for the input!